### ISSUE TITLE
Marvin - Install issue in windows

### ISSUE THREAD

**Issue by margudo** [sdss/marvin#613]
**Describe the bug**

"NoModuleFound" error when importing Marvin in a notebook.

**Desktop (please complete the following information):**

Anaconda install of Jupyter Notebooks, on Windows 10 x64

**Additional context**

Issue via SDSS helpdesk. Additional info from the mail:

To install the library, I ran the pip install command through the normal Windows command shell. It successfully installed, as there's a folder under ~\Anaconda3\Lib\site-packages named "marvin". The contents match the Marvin API uploaded to Github. Marvin is also recognized as a module by pip... But still, I get a "NoModuleFound" error within Jupyter, or trying to import it with a different IDE such as Visual Studio.

So, I followed the error path and discovered that ~\marvin\core\exceptions.py is missing an imported module "pwd". My google research says that "pwd" is a UNIX module that acts as a password interface, and comes with the Python install.

There was an alert on the Marvin documentation that it would not work reliably on Mac OS, but does it require a UNIX only module to be usable? Did I make a mistake somewhere in the install process? Am I misunderstanding something?

**Comment by havok2063** [comment]
The issue is related to #397 , #398, #429, and #590.  The `pwd` package is a Unix and MacOS package.  Marvin should work as expected on Unix or MacOS systems.  We are in the process of finding a solution to this problem for Windows machines.

**Comment by havok2063** [comment]
In the meantime, you can make sure you've followed these instructions for installing on Windows.  https://sdss-marvin.readthedocs.io/en/stable/installation.html#marvin-on-windows

**Comment by TechJoshAccount** [comment]
In exceptions.py, I replaced "import pwd" with "import passlib as pwd", which uses an equivalent library built in to my Anaconda Python install. After following the instructions on marvin-at-windows from the docs, and directing marvin to a HOME folder, it worked in my notebook environment.

I also ran into the other issue you mentioned while importing the library. It attempted to call a file path that didn't exist in the HOME folder, to access the latest drpall file that wasn't downloaded. I'll stay up to date on that fix.

**Comment by WeeBankie** [comment]
Following the Marvin Windows installation instructions at https://sdss-marvin.readthedocs.io/en/stable/installation.html#using-ipython. I have have used `pip install sdss-marvin` and see that the `marvin` package is available through `conda list`. However, I get the `ModuleNotFoundError: No module named 'pwd'` error as reported by others. Is there any workaround for this in Windows/Jupyter Notebook?

```
import marvin
---------------------------------------------------------------------------
ModuleNotFoundError                       Traceback (most recent call last)
<ipython-input-10-86893d82dac6> in <module>
----> 1 import marvin

C:\Users\John\Anaconda3\lib\site-packages\marvin\__init__.py in <module>
     25 # time - 483 ms
     26 from marvin import extern
---> 27 from marvin.core.exceptions import MarvinUserWarning, MarvinError
     28 from brain.utils.general.general import getDbMachine, merge
     29 from brain import bconfig

C:\Users\John\Anaconda3\lib\site-packages\marvin\core\exceptions.py in <module>
     14 
     15 import os
---> 16 import pwd
     17 import sys
     18 import warnings

ModuleNotFoundError: No module named 'pwd'
```

**Comment by joelbrownstein** [comment]
Hi Brian,

Michael Talbot has solved this issue in https://github.com/sdss/marvin/blob/windows/python/marvin/core/exceptions.py

Could you have a look at that, please, and perhaps implement the solution at the same time as introducing the curl access method from sdss_access?

Cheers, Joel

**Comment by havok2063** [comment]
@joelbrownstein Great, thanks.  I've followed the changes a little bit.  I suggest creating a new Pull Request for the `windows` branch similar to the `curl` branch you did for `sdss_access`.  Then I can review all the changes a bit better, in parallel, and make comments.

**Comment by joelbrownstein** [comment]
Pull request created!
https://github.com/sdss/marvin/pull/652


On Apr 1, 2019, at 2:51 AM, Brian Cherinka <notifications@github.com<mailto:notifications@github.com>> wrote:


@joelbrownstein<https://github.com/joelbrownstein> Great, thanks. I've followed the changes a little bit. I suggest creating a new Pull Request for the windows branch similar to the curl branch you did for sdss_access. Then I can review all the changes a bit better, in parallel, and make comments.

**Comment by havok2063** [comment]
This is resolved with the merge of #652.
